gen3_tracker/meta/dataframer.py

```diff
@@ -842,121 +699,43 @@ def create_dataframe(
-
-
-def normalize_coding(resource_dict: dict) -> List[Tuple[str, str]]:
-    def extract_coding(coding_list):
-        # return a concatenated string
-        # return ','.join([coding.get('display', '') for coding in coding_list if 'display' in coding])
-        # or alternatively return an array
-        return [coding.get("display", coding.get("code", "")) for coding in coding_list]
-
-    def find_codings_in_dict(d: dict, parent_key: str = "") -> List[Tuple[str, str]]:
-        codings = []
-        for key, value in d.items():
-            if "valueCodeableConcept" in key:
-                continue
-            if isinstance(value, list):
-                # categories are values not codings in the pivot.
-                if "category" in key:
-                    continue
-                for item in value:
-                    if isinstance(item, dict):
-                        # valudeCodeableConcept is a value in the pivot not a coding
-                        # Check if the dict contains a 'coding' list
-                        if "coding" in item and isinstance(item["coding"], list):
-                            coding_string = extract_coding(item["coding"])
-                            codings.append((coding_string, key))
-                        # Recursively search in the dict
-                        codings.extend(find_codings_in_dict(item, key))
-            elif isinstance(value, dict):
-                # Check if the dict contains a 'coding' list
-                if "coding" in value and isinstance(value["coding"], list):
-                    coding_string = extract_coding(value["coding"])
-                    codings.append((coding_string, key))
-                # Recursively search in the dict
-                codings.extend(find_codings_in_dict(value, key))
-        return codings
-
-    return find_codings_in_dict(resource_dict)
-
-
 def is_number(s):
     """Returns True if string is a number."""
     try:
         int(s)
         return True
     except ValueError:
         return False
+    
+
+def get_subject(db: LocalFHIRDatabase, resource: dict) -> dict:
+    cursor = db.connect()
+    subject_key = resource['subject']['reference']
+    cursor.execute("SELECT * FROM resources WHERE key = ?", (subject_key,))
+    row = cursor.fetchone()
+    assert row, f"{subject_key} not found in database"
+    _, _, raw_subject = row
+    subject = json.loads(raw_subject)
+    return traverse(subject)
+
+ 
+def get_observations_by_focus(db: LocalFHIRDatabase, focus_type: str) -> dict:
+    '''create a dict mapping from focus ID of type focus_type to the associated set of observations'''
+
+    cursor = db.connect()
+    cursor.execute("""
+        SELECT *
+        FROM resources
+        WHERE resource_type = ?
+    """, ("Observation",))
+
+    focus_by_id = defaultdict(list)
+    for _, _, focus_resource in cursor.fetchall():
+        
+        focus_key = json.loads(focus_resource)["focus"][0]["reference"]
```

Comment:
That's gonna error if focus doesn't exist. Need to add if statement somewhere

Reply:
updated